src/trio/_core/_concat_tb.py

```diff
@@ -86,7 +86,9 @@ def copy_tb(base_tb: TracebackType, tb_next: TracebackType | None) -> TracebackT
     def copy_tb(base_tb: TracebackType, tb_next: TracebackType | None) -> TracebackType:
         # tputil.ProxyOperation is PyPy-only, and there's no way to specify
         # cpython/pypy in current type checkers.
-        def controller(operation: tputil.ProxyOperation) -> Any | None:  # type: ignore[no-any-unimported]
+        def controller(  # type: ignore[no-any-unimported]
+            operation: tputil.ProxyOperation,
+        ) -> Any | None:  # noqa: ANN401  # Using Any
```

Comment:
Doesn't `mypy` already warn about using `Any`? Is this rule redundant? (Maybe it covers places `mypy` doesn't?)

Reply:
It raised alerts in several places mypy was fine with and lead to better type annotations for keyword arguments in several places, among other small things.

Reply:
I assume this is because we haven't enabled `disallow_any_explicit`? I'd rather use `mypy` for this than `ruff`.

(not sure if that leads to *more* errors though...)

Reply:
Probably yea

Reply:
Tried adding `disallow_any_explicit`, and it raises 94 new errors. Would be better to handle in another pull request if we handle at all, `Any` can be the best way in some cases.

Reply:
If `ANN401` is less good than `disallow_any_explicit`, then I think we should disable it. Up to you though

Reply:
I think that it would be best to enable more mypy checking, but to save it for another pull request. In other pull request, then disabling `ANN401` might be a good idea, because then there will be duplicate disables for lines matching this where `Any` was required.

Reply:
agreed. disable ANN401 for now and enable `disallow_any_explicit` in a separate PR. Possibly same for `disallow_untyped_defs`.

Reply:
Disabled in 8123b72b